scripts/diagnostic_gen.py

```diff
@@ -28,17 +27,24 @@ def main():
     parser.add_argument("--srcDir", help="Source directory to search for usages")
     parser.add_argument("--incDir", help="Include directory to search for usages")
     parser.add_argument("--docs", action="store_true")
+    parser.add_argument("--diagnostics", help="path to diagnostics file")
     parser.add_argument("--slangBin")
     args = parser.parse_args()
 
-    ourdir = os.path.dirname(os.path.realpath(__file__))
-    inf = open(os.path.join(ourdir, "diagnostics.txt"))
+    inf = None
+    try:
+        inf = open(args.diagnostics)
+    except Exception:
+        print("failed to open", args.diagnostics)
+        sys.exit(1)
 
     headerdir = os.path.join(args.outDir, "slang", "diagnostics")
     try:
-        os.makedirs(headerdir)
-    except OSError:
-        pass
+        os.makedirs(headerdir, exist_ok=True)
+    except OSError as err:
```

Comment:
Same here. The only reason the try block was here originally was in case the directory already exists, which it looks like `exist_ok` will handle now.

Reply:
Sorry, this was where the `pass` was, not above. So the comment above applies here. Exceptions can still happen, so I would keep the try/catch block, and let it buuble up, not pass, nor exit.